packages/nimble-components/src/rich-text/editor/tests/rich-text-editor-mention.spec.ts

```diff
@@ -849,4 +831,550 @@ describe('RichTextEditor user mention via template', () => {
+            ).toEqual(['username2']);
+        });
+
+        it('should get second option as selected on arrow key down', async () => {
+            await appendUserMentionConfiguration(element, [
+                { key: 'user:1', displayName: 'username1' },
+                { key: 'user:2', displayName: 'username2' }
+            ]);
+            await pageObject.setEditorTextContent('@');
+            await pageObject.pressArrowKeyInEditor(ArrowKeyButton.down);
+
+            expect(pageObject.getSelectedOption()).toBe('username2');
+        });
+
+        it('should get first option as selected on arrow key down and up', async () => {
+            await appendUserMentionConfiguration(element, [
+                { key: 'user:1', displayName: 'username1' },
+                { key: 'user:2', displayName: 'username2' }
+            ]);
+            await pageObject.setEditorTextContent('@');
+            await pageObject.pressArrowKeyInEditor(ArrowKeyButton.down);
+            await pageObject.pressArrowKeyInEditor(ArrowKeyButton.up);
+
+            expect(pageObject.getSelectedOption()).toBe('username1');
+        });
+
+        it('should commit second option on arrow key down and enter', async () => {
+            await appendUserMentionConfiguration(element, [
+                { key: 'user:1', displayName: 'username1' },
+                { key: 'user:2', displayName: 'username2' }
+            ]);
+            await pageObject.setEditorTextContent('@');
+            await pageObject.pressArrowKeyInEditor(ArrowKeyButton.down);
+            await pageObject.pressEnterKeyInEditor();
+
+            expect(pageObject.getMarkdownRenderedTagNames()).toEqual([
+                'P',
+                RICH_TEXT_MENTION_USERS_VIEW_TAG
+            ]);
+            expect(
+                pageObject.getEditorMentionViewAttributeValues('mention-label')
+            ).toEqual(['username2']);
+            expect(pageObject.isMentionListboxOpened()).toBeFalse();
+        });
+
+        it('should commit second option on arrow key down and tab', async () => {
+            await appendUserMentionConfiguration(element, [
+                { key: 'user:1', displayName: 'username1' },
+                { key: 'user:2', displayName: 'username2' }
+            ]);
+            await pageObject.setEditorTextContent('@');
+            await pageObject.pressArrowKeyInEditor(ArrowKeyButton.down);
+            await pageObject.pressTabKeyInEditor();
+
+            expect(pageObject.getMarkdownRenderedTagNames()).toEqual([
+                'P',
+                RICH_TEXT_MENTION_USERS_VIEW_TAG
+            ]);
+            expect(
+                pageObject.getEditorMentionViewAttributeValues('mention-label')
+            ).toEqual(['username2']);
+            expect(pageObject.isMentionListboxOpened()).toBeFalse();
+        });
+
+        it('focus out from the editor should close the mention popup', async () => {
```

Comment:
I think we're missing a case to verify that pressing ESC closes the popup

Reply:
Added a test case to check this esc behavior.